lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,133 @@
+      def execute
+        begin
+          namespace = ("HammerCLI" + option_plugin_name.split('_').collect!{ |w| w.downcase.capitalize }.join).constantize
+          options = namespace::Defaults.generate_defaults()
+          added_default_message(options)
+        rescue NameError => e
+          plugin_prob_message
+          HammerCLI::EX_CONFIG
+        rescue StandardError => e
+          file_not_found_message
+          HammerCLI::EX_CONFIG
+        end
+        HammerCLI::EX_OK
+      end
+    end
+
+    def added_default_message(options = {})
+      options.each do |key, val|
+        print_message("Added *" + key.to_s + " default-option with value *"+ val.to_s)
+      end
+    end
+
+    def plugin_prob_message(options = {})
+      print_message("Couldn't reach the plugin defaults class, eiter the plugin doesn't have defaults class or its unreachable, also check for spelling(Snake case)")
+    end
+
+    def file_not_found_message
+      print_message("Couldn't find any cli_config file in the recoginized paths")
+    end
+
+    def list_all_defaults_message
+      HammerCLI::Settings.settings.each do |key,val|
+        if key.to_s.include?("_defaults")
+          print_message('')
+          print_message(key.to_s+":")
+          print_message(YAML.dump(HammerCLI::Settings.settings[key]))
+        end
+      end
+    end
+
+    autoload_subcommands
+  end
+
+
+  class Defaults
+    #this method will be overriden by plugins who wish to have a generate defaults, the plugin will call super(options) on what he wants to generate.
+    def self.generate_defaults(options={})
+      add_defaults_to_conf('global', options)
+      options
+    end
+
+    def self.add_defaults_to_conf(resource, options)
+      path = find_valid_conf_file
+      resource += "_defaults" if resource
+      resource = resource.to_sym
+      if path
+        new_file = YAML.load_file(path)
+        options.each do |key, value|
+          key = key.to_sym
+          new_file[resource] = {} unless new_file[resource]
+          new_file[resource][key] = {} unless new_file[resource][key]
+          value.to_s.to_i if value.is_a? Integer
+          new_file[resource][key] = value
+        end
+        File.open(path,'w') do |h|
+          h.write new_file.to_yaml
+        end
+      else
+        raise StandardError
+      end
+    end
+
+    def self.get_defaults_by_key(key, option)
+      option = option.to_s.sub!('@option_','').to_sym
+      key= (key.to_s + "_defaults").to_sym
+      value = HammerCLI::Settings.settings[key][option] unless HammerCLI::Settings.settings[key].nil?
+      value = HammerCLI::Settings.settings[:global_defaults][option] if value.nil? && !HammerCLI::Settings.settings[:global_defaults].nil?
+      value
+    end
+
+    private
+    def self.find_valid_conf_file
+      path = nil
+      HammerCLI::Settings.path_history.each do |p|
+        path = p if p.include?("cli_config.yml")
```

Comment:
We can also do that, but the thing is that we said that its better the same file.
I think that for now we can leave it like that, if we would like to change it afterwards, its a small fix. 
How common it is for a regular user to manage multiple instances?